Src/main.c

```diff
@@ -108,8 +122,23 @@ int main(void)
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+    // Will range from 0 - 4095
+    adcValue = HAL_ADC_GetValue(&hadc);
+
+    adcPercentage = (float)adcValue/MAXADC;
+
+    //Needs to be a value between 1 and 2ms so the it will be the percentage + 1 (50% is really 1.5ms)
+    //That value needs to then be scalled by the timer period and desired period = 60000/20 = 3000
+    pwmValue = (adcPercentage*(HIGH - LOW) + LOW) * SCALE;
```

Comment:
Try this with `adcValue = 1` and you get `0` as the output when `1` is expected. Remember how C converts `floats` to `ints` (see https://www.cs.cmu.edu/~rbd/papers/cmj-float-to-int.html for some discussion on the topic)

Reply:
Sorry, you get 3000, but 3001 is expected. My bad